tests/tfengine/integration_test.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration_test
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"testing"
+	"text/template"
+	"time"
+
+	"github.com/GoogleCloudPlatform/healthcare-data-protection-suite/internal/tfengine"
+)
+
+var (
+	dirs = []string{"project_secrets"}
+
+	confTmpl = template.Must(template.New("").Parse(`
+template "main" {
+	recipe_path = "{{.CWD}}/../../examples/tfengine/templates/team.hcl"
+	data = {
+		state_bucket    = "placeholder" # Remote backend block will be removed by test.
+		prefix          = "{{.PREFIX}}"
+		billing_account = "{{.BILLING_ACCOUNT}}"
+		parent_id       = "{{.FOLDER_ID}}"
+	}
+}
+`))
+)
+
+func TestFullDeployment(t *testing.T) {
+	if os.Getenv("RUN_INTEGRATION_TEST") == "" {
+		t.SkipNow()
+	}
+
+	tmp, err := ioutil.TempDir("", "")
+	if err != nil {
+		t.Fatalf("ioutil.TempDir = %v", err)
+	}
+	defer os.RemoveAll(tmp)
+
+	data := buildData(t)
+
+	var buf bytes.Buffer
+	if err := confTmpl.Execute(&buf, data); err != nil {
+		t.Fatalf("confTmpl.Execute(&buf, %v) = %v", data, err)
+	}
+
+	confPath := filepath.Join(tmp, "config.hcl")
+	b := buf.Bytes()
+	t.Logf("config file:\n%v", string(b))
+	if err := ioutil.WriteFile(confPath, b, 0644); err != nil {
+		t.Fatalf("ioutil.WriteFile = %v", err)
+	}
+
+	if err := tfengine.Run(confPath, tmp, &tfengine.Options{Format: true, CacheDir: tmp}); err != nil {
+		t.Fatalf("tfengine.Run = %v", err)
+	}
+
+	for _, dir := range dirs {
+		path := filepath.Join(tmp, dir)
+		if err := tfengine.ConvertToLocalBackend(path); err != nil {
+			t.Fatalf("ConvertToLocalBackend(%v): %v", path, err)
+		}
+
+		setupCmd := func(cmd *exec.Cmd) {
+			cmd.Dir = path
+			cmd.Stdout = os.Stdout
+			cmd.Stderr = os.Stderr
```

Comment:
If you add `return cmd` here, you can replace the lower lines with
```
init := setupCmd(exec.Command("terraform", "init"))
apply := setupCmd(exec.Command("terraform", "apply", "-auto-approve"))
destroy := setupCmd(exec.Command("terraform", "destroy", "-auto-approve"))
```

Reply:
another great idea! slightly tweaked